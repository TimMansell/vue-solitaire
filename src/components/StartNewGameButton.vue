<template>
  <Button @click="newGame" to-page="home" data-test="start-new-game-btn">
    New
  </Button>
</template>

<script>
import { mapActions } from 'vuex';
import Button from './Button.vue';

export default {
  name: 'StartNewGameButton',
  components: {
    Button,
  },

  methods: {
    ...mapActions(['newGame']),
  },
};
</script>
