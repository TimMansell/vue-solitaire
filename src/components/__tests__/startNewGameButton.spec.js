import { shallowMount } from '@vue/test-utils';
import StartNewGameButton from '@/components/StartNewGameButton.vue';

describe('StartNewGameButton.vue', () => {
  it('renders the component without crashing', () => {
    const wrapper = shallowMount(StartNewGameButton);

    expect(wrapper.isVisible()).toBe(true);
  });
});
