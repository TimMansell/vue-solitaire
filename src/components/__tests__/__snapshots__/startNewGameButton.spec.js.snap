// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartNewGameButton.vue matches snapshot 1`] = `
<button-stub type="default" size="md" topage="home" data-test="start-new-game-btn">
  New
</button-stub>
`;
